raft/server.cc

```diff
@@ -1055,7 +1072,16 @@ future<> server_impl::applier_fiber() {
                // (i.e. didn't yet receive from _apply_entries queue) but will soon. We avoid unnecessary work
                // of taking snapshots ourselves but comparing our last index directly with what's currently in _fsm.
                auto last_snap_idx = _fsm->log_last_snapshot_idx();
-               if (_applied_idx >= last_snap_idx && _applied_idx - last_snap_idx >= _config.snapshot_threshold) {
+
+               // Having additional room of max_command_memory_usage() ensures
+               // that consume_memory won't get stuck because the last applier_fiber interation
+               // skipped taking a snapshot.
+               const auto memory_limit_approached =
+                   _fsm->log_memory_usage() >= (_config.max_log_size - _config.max_command_memory_usage());
```

Comment:
Why not introduce and use a snapshot threshold in bytes? Why do we have an option to specify the threshold in entires, but don't have one in bytes?

Reply:
Because it's not needed? For `max_log_size` we also have only bytes version here, for simplicity.

Reply:
Different handling of different types of limits (count, bytes) is not simpler, it's more complicated. To fix the bug, it was enough to duplicate all checks, but replace the entry counter with entry size counter.